balancer/state.go

```diff
@@ -0,0 +1,165 @@
+package balancer
+
+import (
+	"context"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"sync/atomic"
+	"time"
+
+	radio "github.com/R-a-dio/valkyrie"
+	"github.com/R-a-dio/valkyrie/config"
+	"github.com/R-a-dio/valkyrie/errors"
+)
+
+// Balancer represents the state of the load balancer.
+type Balancer struct {
+	config.Config
+	manager radio.ManagerService
+	storage radio.StorageService
+	serv    *http.Server
+
+	// The current stream to re-direct clients to.
+	current atomic.Value
+
+	// The amount of listeners from every relay.
+	listeners int
+}
+
+func (br *Balancer) getCurrent() string {
+	return br.current.Load().(string)
+}
+
+// we know statically that current will only be of type string.
+func (br *Balancer) setCurrent(str string) {
+	br.current.Store(str)
+}
+
+// health checks the status of r using c, returning a copy of r.
+func health(ctx context.Context, c *http.Client, r radio.Relay) radio.Relay {
```

Comment:
does this function not eat errors a bit too hastily? maybe print them at the least or return them to be stored in the db somehow

Reply:
An `Err` field was added to the `Relay` type that is saved to the database.